datalab/datalab_session/analysis/raw_data.py

```diff
@@ -22,24 +22,22 @@ def raw_data(input: dict):
     # resize the image to max. 500 pixels on an axis by default for the UI
     max_size = input.get('max_size', 500)
     image = Image.fromarray(image_data)
-    newImage = image.resize((max_size, max_size), Image.LANCZOS)
     bitpix = abs(int(sci_hdu.header.get('BITPIX', 16)))
     max_value = int(sci_hdu.header.get('SATURATE', 0))  # If saturate header is present, use that as max value
-    match bitpix:
-        case 8:
-            datatype = np.uint8
-            if not max_value:
-                max_value = np.iinfo(datatype).max
-        case 16:
-            datatype = np.float16
-            if not max_value:
-                max_value = np.finfo(datatype).max
-        case 32:
-            datatype = np.float32
-            if not max_value:
-                max_value = np.finfo(datatype).max
-    scaled_array = np.asarray(newImage).astype(datatype)
-    scaled_array_flipped = np.flip(scaled_array, axis=0)
+
+    BITPIX_TYPES = {
+        8: (np.uint8, np.iinfo(np.uint8).max),
+        16: (np.float16, np.finfo(np.float16).max),
+        32: (np.float32, np.finfo(np.float32).max)
+    }
+    
+    datatype, max_value_default = BITPIX_TYPES.get(bitpix, (np.uint16, np.iinfo(np.uint16).max))
+    max_value = max_value or max_value_default
+    scaled_image = np.asarray(
+        Image.fromarray(image_data).resize((max_size, max_size), Image.LANCZOS)
+    )
+    # flip the scaled image vertically
+    scaled_array = scaled_image.astype(datatype)[::-1]
```

Comment:
flip already didn't seem to use any memory in my profiling... Did you see it using memory?